Vertexvis/vertex-web-sdk
# [PLAT-1791] Add support for custom scene-tree operation handlers

## Summary

Adds the `selectionHandler`, `expansionHandler`, and `visibilityHandler` properties to the `vertex-scene-tree-table-cell` which, if specified, will override the default selection, expansion, and visibility behaviors respectively. As this provides an alternate approach to what we created the `selectionToggled`, `expansionToggled`, and `visibilityToggled` events for, those events have been removed. Additionally, the `disableInteractions` and `recurseParentSelectionDisabled` flags have been removed in favor of providing custom handlers that modify the behavior.

## Test Plan

- Verify providing a `selectionHandler` with the same selection behavior, and an additional `!event.altKey` check prevents selection when the alt key is pressed
- Verify items are selected while holding the `Alt` key and not setting any `selectionHandler`

## Release Notes

N/A

## Possible Regressions

Selection while holding the `Alt` key and without this new flag set to `true`

## Dependencies

N/A
